cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1479,4 +1492,16 @@ private void resetLogLevels(HttpRequest request, HttpResponder responder, String
     }
     return result;
   }
+
+  private NamespaceId validateAndGetNamespace(String namespace) throws NamespaceNotFoundException {
+    NamespaceId namespaceId = Ids.namespace(namespace);
+    try {
+      namespaceQueryAdmin.get(namespaceId);
+    } catch (NamespaceNotFoundException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
```

Comment:
ok I see, there is a comment there about this catch not being possible. Maybe just put the same comment here.